gradio/components/image.py

```diff
@@ -104,40 +97,34 @@ def __init__(
             render: If False, component will not render be rendered in the Blocks context. Should be used if the intention is to assign event listeners now but render the component later.
             root_url: The remote URL that of the Gradio app that this component belongs to. Used in `gr.load()`. Should not be set manually.
             mirror_webcam: If True webcam will be mirrored. Default is True.
-            brush_radius: Size of the brush for Sketch. Default is None which chooses a sensible default
-            brush_color: Color of the brush for Sketch as hex string. Default is "#000000".
-            mask_opacity: Opacity of mask drawn on image, as a value between 0 and 1.
             show_share_button: If True, will show a share icon in the corner of the component that allows user to share outputs to Hugging Face Spaces Discussions. If False, icon does not appear. If set to None (default behavior), then the icon appears if this Gradio app is launched on Spaces, but not otherwise.
         """
-        self.brush_radius = brush_radius
-        self.brush_color = brush_color
-        self.mask_opacity = mask_opacity
+
         self.mirror_webcam = mirror_webcam
         valid_types = ["numpy", "pil", "filepath"]
         if type not in valid_types:
             raise ValueError(
                 f"Invalid value for parameter `type`: {type}. Please choose from one of: {valid_types}"
             )
         self.type = type
-        self.shape = shape
         self.height = height
         self.width = width
         self.image_mode = image_mode
-        valid_sources = ["upload", "webcam", "canvas"]
-        if source not in valid_sources:
-            raise ValueError(
-                f"Invalid value for parameter `source`: {source}. Please choose from one of: {valid_sources}"
-            )
-        self.source = source
-        if tool is None:
-            self.tool = "sketch" if source == "canvas" else "editor"
-        else:
-            self.tool = tool
-        self.invert_colors = invert_colors
+        valid_sources = ["upload", "webcam", "clipboard"]
+
+        for source in sources:
+            if source not in valid_sources:
+                raise ValueError(
+                    f"Invalid value for parameter `source`: {source}. Please choose any of: {valid_sources}"
+                )
+        self.sources = sources
+
         self.streaming = streaming
         self.show_download_button = show_download_button
-        if streaming and source != "webcam":
-            raise ValueError("Image streaming only available if source is 'webcam'.")
+        if streaming and len(sources) != 1 and sources[0] != "webcam":
+            raise ValueError(
+                "Image streaming only available if sources is ['webcam']. Streaming not supported with multiple sources."
+            )
```

Comment:
new streaming check.

Reply:
Shouldn't it be `not "webcam" in sources` since it could be in any location in the sources array?

Reply:
No, so the idea (for now at least) is that iit must be exactly 'webcam' for streaming to take effect because i doubt people want to handle streams of image snapshot as well as discreet images. Streaming also won't work with other sources types i think (maybe).

So if the sources == `['webcam', 'upload']` then this error should be raised.

Reply:
Ah ok. I think it would be better to rewrite as `if streaming and sources != ['webcam']` then!